app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -18,6 +18,56 @@ $(function () {
     $('#ldap_conn_message').html(msg)
   }
 
+  function ldapConnectionTestPart1 () {
+    var file = $('#dex_connector_ldap_certificate').val()
+    if(!file) {
+      ldapConnectionTestPart2( $('#dex_connector_ldap_current_cert').val() )
+      return;
+    }
+    var fileInput = $('dex_connector_ldap_certificate')[0];
+    var reader = new FileReader();
+
+    reader.onload = function(e) {
+      ldapConnectionTestPart2(reader.result);
+    }
+
+    reader.readAsText(fileInput.files[0]);	
+  }
+
+  function stripCert (cert) {
+    cert = cert.replace("-----BEGIN CERTIFICATE-----\n",'');
```

Comment:
Note that a certificate file could be uploaded without the BEGIN/END blocks. That file would "pass" the connection test button, but would still fail on upload because the backend is expending those to be there. This may be a sub-optimal user experience.

Reply:
Can't we validate the params in the controller to force the user to always upload a valid cert file with the BEGIN/END lines? If they are not present, test connection would fail until the user fixes it. 

I thought that this was already being handled since there's a rescue to `OpenSSL::X509::CertificateError`. Do we need to remove the BEGIN/END lines because of `decoded_pem = Base64.decode64(pem_string)` or something else? In the `x509_certificate_validator` we've been using the whole content of the certificate in the constructor method and it works as expected, no? I just want to understand the reason behind the removal of the BEGIN/END lines. Lemme know if I missed something (I probably did :sweat_smile:).

Reply:
The backend already does enforce that the BEGIN/END blocks are there. All I am saying is that the test connection button currently does not enforce that they are there, it simply removing them before testing the connection as the backend test connection code doesn't work with them present.

The backend of the 'test connection' feature could be changed to remove them at that point. This PR is meant to be as quick as possible to fix the broken head state with minimal change. It serves that purpose and should be accepted and merged. If further changes are desired that is fine but this PR should not be held up for that reason.

Reply:
The BEGIN/END removal has now been moved to the backend to prevent the possible issues described.